cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -241,6 +241,32 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 				Region: options.Region,
 			}))
 
+			osbuildJobResult.Success = true
+			osbuildJobResult.UploadStatus = "success"
+		case *target.AWSS3TargetOptions:
+			a, err := awsupload.New(options.Region, options.AccessKeyID, options.SecretAccessKey)
+			if err != nil {
+				appendTargetError(osbuildJobResult, err)
+				return nil
+			}
+
+			key := options.Key
+			if key == "" {
+				key = uuid.New().String()
+			}
+
+			_, err = a.Upload(path.Join(outputDirectory, exportPath, options.Filename), options.Bucket, key)
+			if err != nil {
+				appendTargetError(osbuildJobResult, err)
+				return nil
+			}
+			url, err := a.S3ObjectPresignedURL(options.Bucket, key)
+			if err != nil {
+				appendTargetError(osbuildJobResult, err)
+			}
```

Comment:
I think we should also `return nil` here. Otherwise, this will look like a success.

Reply:
Thanks for catching this.